src/main/java/org/amoseman/steganography/ui/command/UserInterface.java

```diff
@@ -0,0 +1,92 @@
+package org.amoseman.steganography.ui.command;
+
+/**
+ * Provides a user interface for the program.
+ */
+public class UserInterface {
+    public static final String ARGUMENT_PREFIX = "--";
+    public static final String ENCODE_COMMAND = "encode";
+    public static final String DECODE_COMMAND = "decode";
+    public static final String ANALYZE_COMMAND = "analyze";
+    public static final String SOURCE_ARGUMENT = "source";
+    public static final String TARGET_ARGUMENT = "target";
+    public static final String COMPRESS_ARGUMENT = "compress";
+    public static final String PASSWORD_ARGUMENT = "password";
+
+    private final ArgParser parser;
+
+    /**
+     * Instantiate a user interface.
+     */
+    public UserInterface() {
+        this.parser = new ArgParser.Builder()
```

Comment:
This looks really good!  Nice clean method calls, very easy to read and the intent and action is clear.